backend/src/plugins/Automod/actions/startThread.ts

```diff
@@ -0,0 +1,93 @@
+import { GuildFeature, ThreadAutoArchiveDuration } from "discord-api-types";
+import { TextChannel } from "discord.js";
+import * as t from "io-ts";
+import { renderTemplate, TemplateSafeValueContainer } from "src/templateFormatter";
+import { convertDelayStringToMS, MINUTES, tDelayString, tNullable } from "src/utils";
+import { savedMessageToTemplateSafeSavedMessage, userToTemplateSafeUser } from "src/utils/templateSafeObjects";
+import { noop } from "../../../utils";
+import { automodAction } from "../helpers";
+
+export const StartThreadAction = automodAction({
+  configType: t.type({
+    name: tNullable(t.string),
+    auto_archive: tDelayString,
+    private: tNullable(t.boolean),
+    slowmode: tNullable(tDelayString),
+    limit_per_channel: tNullable(t.number),
+  }),
+
+  defaultConfig: {
+    limit_per_channel: 5,
+  },
+
+  async apply({ pluginData, contexts, actionConfig, ruleName }) {
+    // check if the message still exists, we don't want to create threads for deleted messages
+    const threads = contexts.filter(c => {
+      if (!c.message || !c.user) return false;
+      const channel = pluginData.guild.channels.cache.get(c.message.channel_id);
+      if (channel?.type !== "GUILD_TEXT" || !channel.isText()) return false; // for some reason the typing here for channel.type defaults to ThreadChannelTypes (?)
+      // check against max threads per channel
+      if (actionConfig.limit_per_channel && actionConfig.limit_per_channel > 0) {
+        const threadCount = [
+          ...channel.threads.cache
+            .filter(
+              tr =>
+                tr.ownerId === pluginData.client.application!.id &&
+                !tr.deleted &&
+                !tr.archived &&
+                tr.parentId === channel.id,
+            )
+            .keys(),
+        ].length; // very short line, yes yes
+        if (threadCount >= actionConfig.limit_per_channel) return false;
+      }
+      return channel.messages.cache.has(c.message.id);
+    });
+
+    const guild = await pluginData.guild;
+    const archiveSet = actionConfig.auto_archive
+      ? Math.ceil(Math.max(convertDelayStringToMS(actionConfig.auto_archive) ?? 0, 0) / MINUTES)
+      : 1400;
+    let autoArchive: ThreadAutoArchiveDuration;
+    if (archiveSet === 1440) {
```

Comment:
Make an enum or check if discord.js provides one of their own instead of using magic numbers